internal/pkgecosystem/npm.go

```diff
@@ -29,11 +32,127 @@ func getNPMLatest(pkg string) (string, error) {
 	return details.DistTags.Latest, nil
 }
 
+func makeNpmPackageSpec(pkgName, version string) string {
+	if version == "" {
+		return pkgName
+	} else {
+		return pkgName + "@" + version
+	}
+}
+
+// npmPackData represents the JSON format output by `npm pack`
+// when it successfully downloads a package archive.
+// The format was found by inspecting JSON output of the NPM CLI.
+type npmPackData struct {
+	PackageId    string        `json:"id"`
+	Name         string        `json:"name"`
+	Version      string        `json:"version"`
+	Size         int           `json:"size"`
+	UnpackedSize int           `json:"unpackedSize"`
+	ShaSum       string        `json:"shasum"`
+	Integrity    string        `json:"integrity"`
+	Filename     string        `json:"filename"`
+	Files        []npmFileData `json:"files"`
+	EntryCount   int           `json:"entryCount"`
+	Bundled      interface{}   `json:"bundled"` // not needed
```

Comment:
If it's not needed, can we just omit it completely?

Reply:
I'm not actually sure what it does, but I'm including everything I saw in the NPM output for completeness as it might end up being useful later. Perhaps I should just delete the comment?

Reply:
Will remove all this since I'm going to use direct API requests